Configuration.mk

```diff
@@ -152,11 +152,25 @@ override CPPFLAGS_rv32imac += \
       -Wl,--no-relax   # Prevent use of global_pointer for riscv
 
 override LINK_LIBS_rv32 += \
-      -lc -lgcc -lm -lstdc++ -lsupc++
+      -lgcc -lstdc++ -lsupc++
 
 override LINK_LIBS_rv32imc  += $(LINK_LIBS_rv32)
 override LINK_LIBS_rv32imac += $(LINK_LIBS_rv32)
 
+override LEGACY_LIBS_rv32 += \
+      $(TOCK_USERLAND_BASE_DIR)/newlib/rv32/rv32i/libc.a\
+      $(TOCK_USERLAND_BASE_DIR)/newlib/rv32/rv32i/libm.a
+
+override LEGACY_LIBS_rv32im += \
+      $(TOCK_USERLAND_BASE_DIR)/newlib/rv32/rv32im/libc.a\
+      $(TOCK_USERLAND_BASE_DIR)/newlib/rv32/rv32im/libm.a
+
+override LEGACY_LIBS_rv32imc += $(LEGACY_LIBS_rv32im)
```

Comment:
This isn't great. This means that IMC targets which do support compressed instructions won't use them, bloating the binary and making it slower. Judging from the output that does seem to be the case:

```
Application size report for target rv32imac.0x40440060.0x80007000:
   text	   data	    bss	    dec	    hex	filename
   6816	    148	   2500	   9464	   24f8	build/rv32imac/rv32imac.0x40440060.0x80007000.elf
Application size report for target rv32imc.0x20030080.0x10005000:
   text	   data	    bss	    dec	    hex	filename
   9476	    148	   2500	  12124	   2f5c	build/rv32imc/rv32imc.0x20030080.0x10005000.elf
```

Reply:
If someone can get an `imc` version to build, we could easily update to it.